### ISSUE TITLE
Codegen watch skip first run

### ISSUE THREAD

**Issue by flippidippi** [dotansimha/graphql-code-generator#6803]
**Is your feature request related to a problem? Please describe.**
Currently, we run codegen, then codegen watch to ensure that the first run is successful before compiling the project. However, when the codegen watch is started, it runs an extra codegen before starting the watch.

**Describe the solution you'd like**

It would be nice if we could run the codegen watch and send a flag for it to not run its initial codegen until file changes.

**Describe alternatives you've considered**

No alternatives thought of

**Additional context**

**Comment by ertrzyiks** [comment]
If we agree on the solution, I offer my time to implement it.

**Comment by flippidippi** [comment]
I don't think we can use the current watch flag as it's a boolean or string blob of files. I think we would need a new flag like `--watchSkipFirstCodegen` (idk I'm bad at naming 😄) and we would skip that initial codegen.

https://github.com/dotansimha/graphql-code-generator/blob/bed5bdf04dd18b87490b135467c8a6b96a87d1fe/packages/graphql-codegen-cli/src/utils/watcher.ts#L148

**Comment by jbaubree** [comment]
@dotansimha @charlypoly  any update on this ?

**Comment by flippidippi** [comment]
Anyone try implementing this? @ertrzyiks?

**Comment by eddeee888** [comment]
Sorry this took me awhile to get to.
I think this is a bug, I can't think of a reason why it'd need to run another time immediately. Let me take a look

**Comment by eddeee888** [comment]
I had a look but cannot reproduce the issue in the latest versions.

@flippidippi did you have a reproduction I could use?
